test/suites/edgexfoundry/secrets_config_test.go

```diff
@@ -41,3 +43,26 @@ func TestAddAPIGatewayUser(t *testing.T) {
 	require.NoError(t, err)
 	t.Logf("Output: %s", body)
 }
+
+// TestChangeTlsCert creats new TLS certificate and calls API gateway to verify the use of new certificate
+// https://docs.edgexfoundry.org/3.0/getting-started/Ch-GettingStartedSnapUsers/#changing-tls-certificates
+func TestChangeTlsCert(t *testing.T) {
+	t.Log("Create TLS certificate")
+	createTlsCert(t)
+
+	const coreDataEndpoint = "https://localhost:8443/core-data/api/v2/ping"
+	t.Log("Calling API gateway using new TLS certificates:", coreDataEndpoint)
+	// Note: %%	is a literal percent sign
+	code, _, _ := utils.Exec(t, fmt.Sprintf("curl --show-error --silent --include"+
+		"--output /dev/null --write-out '%%{http_code}' --caCert ca.cert '%s'",
+		coreDataEndpoint))
```

Comment:
This code uses a file that was created in a script, far away from here, which can change without knowing it is used here. I wonder if we can make this less fragile.

Some comments would be helpful.